src/forwardproblem.cpp

```diff
@@ -288,8 +288,12 @@ void ForwardProblem::storeEvent() {
 }
 
 void ForwardProblem::handleDataPoint(int it) {
-    timepoint_states.push_back(getSimulationState());
+    timepoints.push_back(getTime());
     solver->storeDiagnosis();
+    /* We only store the simulation state if it's not the initial state, as the
+       initial state is stored anyway and we want to avoid storing it twice */
+    if (t != model->t0())
+        timepoint_states.insert(std::make_pair(t, getSimulationState()));
```

Comment:
```suggestion
        timepoint_states[t] = getSimulationState();
```
simpler? :)

Reply:
Simpler, but `insert` checks for existence of that key, whereas `[t]` overwrites the entry if existing. SO I would prefer to go with `insert`, as this avoids writing the same content multiple times, if fine. :)

Reply:
Good point. But then we would want to skip creation of a new SimulationState instance (i.e. not calling getSimulationState()) which will still happen. Overriding it in the map is way cheaper than that...